infrastructures/kubernetes/src/test/java/org/eclipse/che/workspace/infrastructure/kubernetes/provision/GitUserProfileProvisionerTest.java

```diff
@@ -128,4 +152,67 @@ public void testShouldCheckIfPodHasMountAndK8HasConfigMapForGitConfig() throws E
 
     assertEquals(gitconfig, expectedGitconfig);
   }
+
+  @Test
+  public void
+      testShouldCheckIfPodHasMountAndK8HasConfigMapForGitConfigForDifferentTypeOfPreference()
+          throws Exception {
+    String json =
+        "{\"chePlugins.repositories\":{\"Plugins\":\"http://url/\",\"my\":\"https://url/plugins.json\"}, \"git.user.name\":\"user\",\"git.user.email\":\"email\"}";
+    Map<String, String> preferences = singletonMap("theia-user-preferences", json);
+    when(preferenceManager.find(eq("id"), eq("theia-user-preferences"))).thenReturn(preferences);
+    when(runtimeIdentity.getWorkspaceId()).thenReturn("wksp");
+
+    ObjectMeta podMeta = new ObjectMetaBuilder().withName("wksp").build();
+    when(pod.getMetadata()).thenReturn(podMeta);
+    when(pod.getSpec()).thenReturn(podSpec);
+    when(podSpec.getContainers()).thenReturn(singletonList(container));
+
+    List<VolumeMount> volumeMounts = new ArrayList<>();
+
+    when(container.getVolumeMounts()).thenReturn(volumeMounts);
+    k8sEnv.addPod(pod);
+
+    gitUserProfileProvisioner.provision(k8sEnv, runtimeIdentity);
+
+    assertEquals(volumeMounts.size(), 1);
+
+    VolumeMount mount = volumeMounts.get(0);
+
+    assertEquals(mount.getMountPath(), "/etc/gitconfig");
+    assertEquals(mount.getName(), "gitconfigvolume");
+    assertFalse(mount.getReadOnly());
+    assertEquals(mount.getSubPath(), "gitconfig");
+
+    assertEquals(k8sEnv.getConfigMaps().size(), 1);
+    assertTrue(k8sEnv.getConfigMaps().containsKey("wksp-gitconfig"));
+
+    ConfigMap configMap = k8sEnv.getConfigMaps().get("wksp-gitconfig");
+
+    assertEquals(configMap.getData().size(), 1);
+    assertTrue(configMap.getData().containsKey("gitconfig"));
+
+    String gitconfig = configMap.getData().get("gitconfig");
+    String expectedGitconfig = "[user]\n\tname = user\n\temail = email\n";
+
+    assertEquals(gitconfig, expectedGitconfig);
+  }
+
+  @Test(expectedExceptions = JsonSyntaxException.class)
+  public void testShouldFailedWhenJsonParserIsConfiguredOnParsingMapStringToString() {
+    // json taken from https://github.com/eclipse/che/issues/14436
+    String json =
+        "{\"chePlugins.repositories\":{\"Eclipse Che plugins\":\"http://plugin-registry-che.192.168.99.112.nip.io/v3/plugins/\",\"my\":\"https://raw.githubusercontent.com/vitaliy-guliy/che-theia-plugin-registry/master/plugins/plugins.json\"}}";
+    Type stringMapType = new TypeToken<Map<String, String>>() {}.getType();
+    new Gson().fromJson(json, stringMapType);
+  }
+
+  @Test
+  public void testShouldPassWhenJsonParserIsConfiguredOnParsingMapStringToObject() {
```

Comment:
These test cases don't seem to test anything in `GitUserProfileProvisioner` -- they're just testing that `Gson().fromJson()` throws the expected error.

The case of `GitUserProfileProvisioner` not handling Objects is covered by the `testShouldCheckIfPodHasMountAndK8HasConfigMapForGitConfigForDifferentTypeOfPreference` test.

Reply:
>These test cases don't seem to test anything in GitUserProfileProvisioner -- they're just testing that Gson().fromJson() throws the expected error.

Exactly, this is kind of regression test. To show that we had such problem.

Reply:
>The case of GitUserProfileProvisioner not handling Objects is covered by the testShouldCheckIfPodHasMountAndK8HasConfigMapForGitConfigForDifferentTypeOfPreference test.

Can you more clarify this? I'm not sure that I understand the main idea.

Reply:
I mean that since these tests only call `Gson#fromJson`, they will never fail regardless of what happens in the Che codebase. The regression test should occur in the other test case (or a new one should be added).

Reply:
Ahh, I see what you mean, so, this tests are redundant and can be removed.